python/ray/train/torch/train_loop_utils.py

```diff
@@ -632,11 +637,20 @@ def __init__(
         self._dataloader = base_dataloader
         self.dataloader_iter = None
         self.device = device
+
+        self.device_manager = get_session().device_manager
+        # reset device is needed for npu in a new thread so far.
+        if device.type == "npu":
+            self.device_manager.set_device(device)
+
         # disable auto transfer (host->device) if cpu is used
-        self._auto_transfer = auto_transfer if device.type == "cuda" else False
-        # create a new CUDA stream to move data from host to device concurrently
+        if device.type == "cuda" or device.type == "npu":
```

Comment:
It's not good to enumerating device type here. To make it generalizable, let's check if the device manager can create a stream instead.

Reply:
Exactly! Enumerating device type here conflict the original purpose of this PR.
Introduced `is_support_stream` API to check it.